submission/programs/adlbc.R

```diff
@@ -0,0 +1,227 @@
+    x == "CL" ~ 20,
+    x == "BILI" ~ 21,
+    x == "ALP" ~ 22,
+    x == "GGT" ~ 23,
+    x == "ALT" ~ 24,
+    x == "AST" ~ 25,
+    x == "BUN" ~ 26,
+    x == "CREAT" ~ 27,
+    x == "URATE" ~ 28,
+    x == "PHOS" ~ 29,
+    x == "CA" ~ 30,
+    x == "GLUC" ~ 31,
+    x == "PROT" ~ 32,
+    x == "ALB" ~ 33,
+    x == "CHOL" ~ 34,
+    x == "CK" ~ 35
+  )
+}
+# Add supplemental information --------------------------------------------
+sup <- supplb %>%
+  select(STUDYID, USUBJID, IDVAR, IDVARVAL, QNAM, QLABEL, QVAL) %>%
+  pivot_wider(
+    id_cols = c(STUDYID, USUBJID, IDVARVAL),
+    names_from = QNAM,
+    values_from = QVAL
+  ) %>%
+  mutate(LBSEQ = as.numeric(IDVARVAL)) %>%
+  select(-IDVARVAL)
+
+adlb00 <- lb %>%
+  left_join(sup, by = c("STUDYID", "USUBJID", "LBSEQ")) %>%
+  filter(LBCAT == "CHEMISTRY")
+
+# ADSL information --------------------------------------------------------
+
+adsl <- adsl %>%
+  select(
+    STUDYID, SUBJID, USUBJID, TRT01PN, TRT01P, TRT01AN, TRT01A, TRTSDT, TRTEDT, AGE, AGEGR1, AGEGR1N, RACE, RACEN, SEX,
+    COMP24FL, DSRAEFL, SAFFL
+  )
+
+
+
+adlb01 <- adlb00 %>%
+  left_join(adsl, by = c("STUDYID", "USUBJID"))
+
+# Dates -------------------------------------------------------------------
+
+adlb02 <- adlb01 %>%
+  derive_vars_dtm(
+    new_vars_prefix = "A",
+    dtc = LBDTC,
+    highest_imputation = "s",
+    ignore_seconds_flag = T
+  ) %>%
+  derive_vars_dt(
+    new_vars_prefix = "A",
+    dtc = LBDTC,
+    highest_imputation = "n"
+  ) %>%
+  derive_vars_dy(reference_date = TRTSDT, source_vars = vars(ADT))
+
+
+
+# AVAL(C) -----------------------------------------------------------------
+# No imputations are done for values below LL or above UL
+
+adlb03 <- adlb02 %>%
+  mutate(
+    AVAL = LBSTRESN,
+    AVALC = ifelse(!is.na(AVAL), LBSTRESC, NA)
+  )
+
+# Parameter ---------------------------------------------------------------
+
+adlb04 <- adlb03 %>%
+  mutate(
+    PARAM = paste0(LBTEST, " (", LBSTRESU, ")"),
+    PARAMCD = LBTESTCD,
+    PARAMN = format_paramn(LBTESTCD),
+    PARCAT1 = "CHEM" # changed to match prod dataset
+  )
+
+# Baseline ----------------------------------------------------------------
+## updating to use admiral programming
+
+
+adlb05 <- adlb04 %>%
+  mutate(ABLFL = LBBLFL) %>%
+  derive_var_base(
+    by_vars = vars(STUDYID, USUBJID, PARAMCD),
+    source_var = AVAL,
+    new_var = BASE
+  ) %>%
+  derive_var_chg()
+
+
+# VISITS ------------------------------------------------------------------
+
+eot <- adlb05 %>%
+  filter(ENDPOINT == "Y") %>%
+  mutate(
+    AVISIT = "End of Treatment",
+    AVISITN = 99
+  )
+
+adlb06 <- adlb05 %>%
+  filter(grepl("WEEK", VISIT, fixed = TRUE) |
+    grepl("UNSCHEDULED", VISIT, fixed = TRUE) |
+    grepl("SCREENING", VISIT, fixed = TRUE)) %>% # added conditions to include screening and unscheduled visits
+  mutate(
+    AVISIT = case_when(
+      ABLFL == "Y" ~ "Baseline",
+      grepl("UNSCHEDULED", VISIT) == TRUE ~ "",
```

Comment:
NA_character_ instead of ""?

Reply:
@SHAESEN2 can we resolve this now?

Reply:
As the QC against the original dataset is passing now, I will go ahead and resolve this for the time being. @SHAESEN2 please follow up if any further questions.